Source/Fuse.Scripting.JavaScript/ObservableProperty.uno

```diff
@@ -40,16 +41,95 @@ namespace Fuse.Scripting.JavaScript
 
 		Observable _observable;
 
+		// JS thread
 		internal Observable GetObservable(Scripting.Context context)
 		{
 			if (_observable == null)
 			{
 				_observable = Observable.Create(context, _worker);
+				_observable.Object["_defaultValueCallback"] = (Scripting.Callback)DefaultValueCallback;
 				Subscribe(context);
 			}
 			return _observable;
 		}
 
+		// JS thread
+		object DefaultValueCallback(Scripting.Context context, object[] args)
+		{
+			var value = args[0];
+
+			object marshalledValue;
+			if(!Marshal.TryConvertTo(_property.PropertyType, value, out marshalledValue))
+			{
+				return null;
+			}
+
+			var resolveClosure = new ResolveDefaultValueClosure(this, marshalledValue);
+			UpdateManager.PostAction(resolveClosure.Perform);
+
+			return null;
+		}
+
+		internal class ResolveDefaultValueClosure
+		{
+			readonly Uno.UX.Property _property;
+			readonly ISubscription _subscription;
+			readonly object _value;
+			readonly Action<object> _pushValueOnJSThread;
+
+			public ResolveDefaultValueClosure(ObservableProperty op, object value)
+			{
+				_property = op._property;
+				_subscription = op._subscription;
+				_pushValueOnJSThread = op.PushValueOnJSThread;
+				_value = value;
+			}
+
+			// UI thread
+			public void Perform()
+			{
+				// HACK: https://github.com/fusetools/fuselibs-public/issues/541#issuecomment-335101235
+				if(IsDefaultValueForType(_property.GetAsObject(), _property.PropertyType))
+				{
+					_property.SetAsObject(_value, null);
+					_pushValueOnJSThread(_value);
+				}
+			}
+
+			static bool IsDefaultValueForType(object value, Type t)
+			{
+				return (!t.IsValueType && value == null)
+					|| IsDefault<bool>(value, t)
+					|| IsDefault<byte>(value, t)
+					|| IsDefault<sbyte>(value, t)
+					|| IsDefault<char>(value, t)
+					|| IsDefault<short>(value, t)
+					|| IsDefault<ushort>(value, t)
+					|| IsDefault<int>(value, t)
+					|| IsDefault<uint>(value, t)
+					|| IsDefault<long>(value, t)
+					|| IsDefault<ulong>(value, t)
+					|| IsDefault<float>(value, t)
+					|| IsDefault<double>(value, t)
+					|| IsDefault<int2>(value, t)
+					|| IsDefault<int3>(value, t)
+					|| IsDefault<int4>(value, t)
+					|| IsDefault<byte2>(value, t)
+					|| IsDefault<byte4>(value, t)
+					|| IsDefault<float2>(value, t)
+					|| IsDefault<float3>(value, t)
+					|| IsDefault<float4>(value, t)
+					|| IsDefault<Size>(value, t)
+					|| IsDefault<Size2>(value, t);
```

Comment:
Yikes! This seems like a maintenance nightmare. When do we need to add types here?!

Reply:
@sebbert